doc/radosgw/rgw-cache.rst

```diff
@@ -9,148 +9,142 @@ RGW Data caching and CDN
-https://www.nginx.com/blog/smart-efficient-byte-range-caching-nginx/
-Using this API gives the ability to read ahead objects when clients asking a specific range from the object.
-On subsequent accesses to the cached object, Nginx will satisfy requests for already-cached ranges from the cache. Uncached ranges will be read from RGW (and cached).
+- **Auth API:** The cache uses this to validate that a user can access the cached data.
+- **Cache API:** Adds the ability to override securely ``Range`` header so that Nginx can use its own `smart cache <https://www.nginx.com/blog/smart-efficient-byte-range-caching-nginx/>`_ on top of S3.
+  Using this API gives the ability to read ahead objects when client is asking a specific range from the object.
+  On subsequent accesses to the cached object, Nginx will satisfy requests for already-cached ranges from the cache. Uncached ranges will be read from RGW (and cached).
 
 Auth API
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~
 
-This API Validates a specific authenticated access being made to the cache, using RGW's knowledge of the client credentials and stored access policy.
+This API validates a specific authenticated access being made to the cache, using RGW's knowledge of the client credentials and stored access policy.
 Returns success if the encapsulated request would be granted.
 
 Cache API
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~~
 
-This API is meant to allow changing signed Range headers using a privileged user, cache user.
+This API is meant to allow changing signed ``Range`` headers using a privileged cache user.
 
-Creating cache user
+Creating the cache user:
 
-::
+.. prompt:: bash #
+
+   radosgw-admin user create --uid=<uid for cache user> --display-name="cache user" --caps="amz-cache=read"
+
+This user can send to the RGW the Cache API header ``X-Amz-Cache``. This header contains the headers from the original request (before changing the ``Range`` header):
 
-$ radosgw-admin user create --uid=<uid for cache user> --display-name="cache user" --caps="amz-cache=read"
+- Original headers are separated from each other by a character with ASCII code 177.
+- Each original header and its value are separated by a character with ASCII code 178.
 
-This user can send to the RGW the Cache API header ``X-Amz-Cache``, this header contains the headers from the original request(before changing the Range header).
-It means that ``X-Amz-Cache`` built from several headers.
-The headers that are building the ``X-Amz-Cache`` header are separated by char with ASCII code 177 and the header name and value are separated by char ASCII code 178.
-The RGW will check that the cache user is an authorized user and if it is a cache user,
-if yes it will use the ``X-Amz-Cache`` to revalidate that the user has permissions, using the headers from the X-Amz-Cache.
-During this flow, the RGW will override the Range header.
+The RGW will check that the cache user is an authorized user and that it is a cache user.
```

Comment:
This is a bit awkward, we call it a cache user then say that we check that it's a cache user.  Maybe something like saying that the value is a cache user?

Reply:
Much appreciated.

Reply:
The original not at all clear sentence could be understood in 2 different ways. Is it trying to say that that there are 2 checks ("is the user authorized?" and "is it a cache user?") or is it just a mangled sentence trying to say simply that it checks that the "cache user is authorized"?  
I had previously assumed it is the previous but given the language I'm not not sure if the fragment "and if it is a cache user, if yes" just repeats the same thing 2 times.

As you say, it makes no sense to first talk about "cache user" but then "check if it is a cache user".

Hmm. Will think about this for a while.